### PR TITLE
Equal split on insertion

### PR DESCRIPTION
**Pull request by facontidavide**
Hi,

I have an application where I split an existing area in two. With this modifications it look much better ;)

See video here: https://vimeo.com/426505015

Actually, it would also be quite easy to animate such "split"... I tried on another branch if you might be interested to merge.

### CODE REVIEW

File: demo/MainWindow.cpp [hunk 1]
```diff
@@ -586,7 +586,7 @@ CMainWindow::CMainWindow(QWidget *parent) :
 
 	d->createContent();
 	// Default window geometry - center on screen
-    resize(1280, 720);
+    resize(1200, 1000);
```
Comment (facontidavide) [inline demo/MainWindow.cpp:589]: to better appreciate the difference before and after the PR

**Comment by dail8859** [comment]
I would also find this beneficial as well since it is clear to the users exactly what will happen when splitting areas.

Currently I'm not sure how it determines how exactly to split areas.

**Comment by githubuser0xFFFF** [comment]
Thank you for your pull request. I just reviewed your code and I agree - for your application it definitely looks nicer. But because I do not know if all people like this change, I made it configurable with the config flag `CDockManager::EqualSplitOnInsertion` that is disabled by default.

I'm interested in your branch where the insertion is animated.

**Comment by dail8859** [comment]
One minor thing I noticed, the blue "drop preview" (not sure the correct terminology) is a bit off as i would expect it to highlight exactly half. Here is a screenshot of the `deleteonclose` example using 58744408f04c446ad0b7c5148944809e9d8d6daf and I have enabled `EqualSplitOnInsertion`

If you have two tabs and drag one to the right to split it then the red line in the screenshot is where I would expect the blue box to start:

image

**Comment by githubuser0xFFFF** [comment]
Equal split on insertion us for runtime insertion via code - not for interactive drag and droo

**Comment by dail8859** [comment]
@githubuser0xFFFF Thanks for the info. Since this is expected behavior then that works for me! 👍

**Pull request merged**